lightning/src/ln/channel.rs

```diff
@@ -2292,7 +2292,22 @@ impl<SP: Deref> InitialRemoteCommitmentReceiver<SP> for FundedChannel<SP> where
 	}
 }
 
-impl<SP: Deref> PendingV2Channel<SP> where SP::Target: SignerProvider {
+// TODO Naming
+pub(super) trait PendingV2ChannelTrait<SP: Deref> where SP::Target: SignerProvider {
+	fn context(&self) -> &ChannelContext<SP>;
+	fn context_mut(&mut self) -> &mut ChannelContext<SP>;
+	fn funding(&self) -> &FundingScope;
+	fn funding_mut(&mut self) -> &mut FundingScope;
+	fn funding_and_context_mut(&mut self) -> (&mut FundingScope, &mut ChannelContext<SP>);
+	fn dual_funding_context(&self) -> &DualFundingChannelContext;
```

Comment:
Do we only need to use `dual_funding_context` in `begin_interactive_funding_tx_construction`? If so, seems like it should be passed into that and consumed rather than needing a method. Then you wouldn't need `swap_out_dual_funding_context_inputs` either.

In other words, can `DualFundingChannelContext` be constructed immediately before calling `begin_interactive_funding_tx_construction` rather than being a field on each channel type?